RTObjCInterop/RTObjCInterop/src/RTObject.mm

```diff
@@ -0,0 +1,55 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+// This has to be @impled in a .mm file because we need to create the ivars for the comptr.
+
+#import <UWP/RTHelpers.h>
+#import <UWP/interopBase.h>
+
+#include <Windows.Foundation.h>
+
+#import <Starboard.h>
+
+@implementation RTObject
++ (instancetype)alloc {
+    NSLog(@"Cannot allocate WinRT components through alloc/new! Use static instantiators.", NSStringFromClass(self));
+    assert(0);
```

Comment:
as are rule, we aren't using asserts/nslog in the framework.  Throwing exception instead may be appropriate here.